R/tag_examplesShinylive.R

```diff
@@ -0,0 +1,202 @@
+#' Custom `examplesShinylive` tag.
+#'
+#' This function generates a new "Examples in Shinylive" section in the documentation. This section contains URL to
+#' the application in Shinylive as well as an iframe with the application.
+#' If no code is provided then the code is taken from the following `@examples` or `@examplesIf` tag.
+#'
+#' The application code must be executable inside Shinylive. If the application code includes functions from your
+#' package, you must add `library(<your package>)` beforehand. For more information, refer to the Decoration section
+#' on how to use and decorate existing examples.
+#'
+#' Note: All the packages used in the application code need to be installable in WebR.
+#' See [this article](https://docs.r-wasm.org/webr/latest/packages.html) for more details.
+#'
+#' @section Decoration:
+#'
+#' To avoid repetition between the `@examplesShinylive` and `@examples` sections, there are special string literals
+#' that provide access to the other tags from within the content of `@examplesShinylive`.
+#' These literals should be used as expressions embraced with `{{ }}`, which are then interpolated using
+#' `glue::glue_data(..., .open = "{{", .close = "}}")`.
+#'
+#' The following keywords are available:
+#' * `"{{ tags_examples }}"` - a list of `@examples` or `@examplesIf` tags
+#' * `"{{ examples }}"` - a list of "raw" elements from `tags_examples` list elements
+#' * `"{{ next_example }}"` - "raw" element of the next example
+#' * `"{{ prev_example }}"` - "raw" element of the previous example
+#'
+#' This allows you to access and decorate existing example code to create executable application code for Shinylive.
+#' Refer to the examples section for possible use cases.
+#'
+#' @name tag-examplesShinylive
+#'
+#' @usage
+#' #' @examplesShinylive${1:# example code (optional)}
```

Comment:
I thought you always need a new line like this:
```
#' @examplesShinylive
#' library(teal)
#' app <- ...
```

Is `${` a special roxygen syntax if including the content on the same line or is this something specific to roxy.shinylive?

Reply:
This is inspired by the roxygen' `@examples` documentation: https://github.com/r-lib/roxygen2/blob/9652d15221109917d46768e836eaf55e33c21633/man/tags-rd.Rd#L24

It looks like that's the way to document the usage section